### PR TITLE
upgrade arc_ec to v0.4

### PR DESCRIPTION
**Pull request by nikkolasg**
## Description

Bring the crate to use ark_ec v0.4

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [] Wrote unit tests
- [] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by nikkolasg** [comment]
BTW we also have a preliminary [branch](https://github.com/cryptonetlab/ark-polycommit/pull/1/) to do PST on the other base - commit in G2 opening in G1 - if you are interested

**Comment by maramihali** [comment]
had a look at the multilinear PST, lgtm

File: src/sonic_pc/data_structures.rs [hunk 1]
```diff
@@ -169,71 +172,77 @@ impl<E: PairingEngine> VerifierKey<E> {
...
+            + self.h.serialized_size(compress)
+            + self.beta_h.serialized_size(compress)
+            + self
+                .degree_bounds_and_neg_powers_of_h
+                .serialized_size(compress)
+            + self.supported_degree.serialized_size(compress)
+            + self.max_degree.serialized_size(compress)
     }
 
+    fn serialize_compressed<W: Write>(&self, writer: W) -> Result<(), SerializationError> {
```
Comment (mmagician) [inline src/sonic_pc/data_structures.rs:214]: ditto

**Review by mmagician — CHANGES_REQUESTED** [review]
mostly looks good! I left a few cleanup comments

**Comment by Pratyush** [comment]
I moved the commits to #112, which is merged now; thanks everyone for the work!

**Pull request closed without merging**